core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/SonarLintPathPattern.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.analysis;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.utils.PathUtils;
+import org.sonar.api.utils.WildcardPattern;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+/**
+ * Path relative to module basedir
+ */
+public class SonarLintPathPattern {
+
+  private static final Logger LOG = Loggers.get(SonarLintPathPattern.class);
+
+  final WildcardPattern pattern;
+
+  public SonarLintPathPattern(String pattern) {
+    if (pattern.startsWith("file:")) {
+      LOG.warn("Unsupported path pattern: " + pattern);
+      pattern = pattern.substring("file:".length());
```

Comment:
If the pattern starts with `file:`, it's likely to have multiple `/` after `file:`, no? In which case, wouldn't it be good to strip those here, that is `pattern = pattern.replaceAll("^file:/*", "")`